internal/compiler/typeregister.rs

```diff
@@ -87,6 +89,7 @@ pub(crate) const RESERVED_DROP_SHADOW_PROPERTIES: &[(&str, Type)] = &[
 pub(crate) const RESERVED_ACCESSIBILITY_PROPERTIES: &[(&str, Type)] = &[
     //("accessible-role", ...)
     ("accessible-checked", Type::Bool),
+    ("accessible-delegate-focus", Type::String),
```

Comment:
Not sure I understand how this works: is this meant to be an int? Which is the child number. This seems quite error prone...  
This need to be documented.
But we probably should discuss another way.

Reply:
A int auto-converts to String...

Yes, this is really ugly and should be improved upon:-) I wanted to demonstrate that it can implement the desired behavior first before discussing how to make this nicer.

Reply:
The auto-converts is not what worries me.
It is the fact that this is a child number, which is totally unreliable, and kind of work by luck for the tab widget but might not.

Reply:
It's worse: It is the child of the accesisble item that is referenced here, not the children of the item :-)

Reply:
Because that requires a way to retrieve the label of the currently focused tab. That in turn requires some hacking in the compiler to provide that information -- which is totally specific to the tab widget.

Then I wanted to add the description, which requires a repeat of that hack.

This way the *entire* set of accessible properties of the child are exposed automatically. And the mechanism can be reused in other situations.

Reply:
> That in turn requires some hacking in the compiler to provide that information

Did you not already do that anyway in a previous iteration?

> which is totally specific to the tab widget.

Well, yes, we have code in the compiler to handle the tab widget, i don't see the problem with that.

> Then I wanted to add the description, which requires a repeat of that hack.

The compiler can just generate the `accessible-label` and `accessible-description` for the tab bar. that seems fairly easy to do. And i don't see that as a "hack".

> This way the entire set of accessible properties of the child are exposed automatically. And the mechanism can be reused in other situations.

But is also specific to the current implementation of the tab widget. and is quite unreliable. You're making a public property just for that.

Reply:
> Did you not already do that anyway in a previous iteration?

Yes, I did. It is complex code that is needs to be done for each property we might want to expose. Code that will just get more complex when we add dynamic tabs...

> But is also specific to the current implementation of the tab widget. and is quite unreliable.

What makes you think this is unreliable?

I expect this to work, even if we add dynamic tabs and stuff. And we do not need to update anything when we need to make more properties accessible.

> You're making a public property just for that.

I do hope to be able to reuse this. It can be used wherever a known set of children is available. E.g. in a ListView or even just to switch between two different objects based on some state.

Reply:
> You will have to explain to me, and in the documentation of the accessible-delegate-focus, how this is supposed to work.

The entire accessibility backend assumes a tree of accessible widgets, with each widget having `childCount` children (in Qt) and then access those children with `child(int)`. So this has to work reliably or the entire accessibility will not work.

> It looks like you're taking the tab number and that you hope that they map to the child of the tab widget.

Almost: I map that to the _accessible_ child of the _accessible_ widget that is the TabBar.

> But this may not even be the case. There may be layout that may or may not be optimized away, there might be other intermediate items coming in (Clip, Opacity, ....) it may be different from style to style.

Any optimization that removes user-visible items is buggy. Any optimization that changes the structure between user visible items (e.g turns some other user visible item from a sibling to a child or switches parent and child) to change is also buggy.

All accessible items are by definition user-visible. So the tree of _accessible_ items should be stable.

Much of the accessibility will be defined inside the styles. Our styled tab bar knows that its children will be tabs and how exactly those are represented. Yes, you will need to make sure no extra children will be sneaked into the accessibility tree below the Tab widget. But making sure widgets work as expected is what a style developer needs to do.

Reply:
> The entire accessibility backend assumes a tree of accessible widgets, with each widget having childCount children (in Qt) and then access those children with child(int). So this has to work reliably or the entire accessibility will not work.

Ah ok, so there was my misunderstanding. I did not really this was a child in the "accessible widget tree", and thought it was a child in the item tree.  That makes it slightly better.

> Any optimization that removes user-visible items is buggy

No. It could merge two items in one for example.  It can also change the order and do more stuff.  (the z order pass changes the order of items)

Reply:
We had a discussion in Mattermost about this: We agreed to keep it as is for now, but to keep the `accessible-delegate-focus` as private implementation detail for now.